### PR TITLE
Feature/use server common pkg

### PR DESCRIPTION
**Pull request by prunelsigma**
- regroup code and use `server-common-go` packages
https://github.com/cryptogarageinc/server-common-go

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -4,32 +4,38 @@ go 1.14
 
 require (
 	github.com/Bose/go-gin-logrus v1.0.3
+	github.com/armon/consul-api v0.0.0-20180202201655-eb2c6b5be1b6 // indirect
```
Comment (Tibo-lg) [inline go.mod]: I know it's an indirect dependency, but apparently this is deprecated, do you know where the dependency comes from?

Comment (Tibo-lg) [reply go.mod]: Actually there are quite more dependencies, is it from the common packages?

File: internal/api/asset_controller.go [hunk 2]
```diff
@@ -87,7 +87,7 @@ func (ct *AssetController) GetAssetRvalue(c *gin.Context) {
 		c.Error(err)
 		return
 	}
-	c.JSON(http.StatusOK, NewDLCDataReponse(oracleInstance.PublicKey, dlcData))
+	c.JSON(http.StatusOK, NewDLCDataResponse(oracleInstance.PublicKey, dlcData))
```
Comment (Tibo-lg) [inline internal/api/asset_controller.go:90]: Am I crazy or there is not diff here?

**Review by Tibo-lg — COMMENTED** [review]
Mostly looks good, just a bit worried about the extra dependencies.

File: internal/api/asset_controller.go (hunk 2 above)
Comment (prunelsigma) [reply internal/api/asset_controller.go:90]: a small `Reponse` -> `Response`
while updating code I saw this, so I though may as well update :)

Comment (Tibo-lg) [reply internal/api/asset_controller.go:90]: Oh right thanks!

File: go.mod (hunk 1 above)
Comment (prunelsigma) [reply go.mod]: hmm some strange things happens I'm not sure but I think `go` try to check for matching dependencies automatically when it does not find in the `go.mod`
but `go mod tidy` fixed this issue, you should have less dependencies now :)
good catch 👍 
I will check also the `server` repository for this problem

**Review by Tibo-lg — APPROVED** [review]

**Pull request merged**